chromium_src/components/printing/renderer/print_render_frame_helper.cc

```diff
@@ -52,6 +52,13 @@ void PrintRenderFrameHelper::InitiatePrintPreview(
     print_preview_context_.InitWithFrame(frame);
     print_preview_context_.DispatchBeforePrintEvent(
         weak_ptr_factory_.GetWeakPtr());
+
+    // Note: |DispatchBeforePrintEvent| and |DispatchAfterPrintEvent| must be
+    // called in pairs.
+    // They're intended to be called before and after the `print` event is
+    // dispatched, but we just call |DispatchAfterPrintEvent| here to avoid a
+    // DCHECK next time we call |DispatchBeforePrintEvent|.
+    print_preview_context_.DispatchAfterPrintEvent();
```

Comment:
@darkdh without this call to `DispatchAfterPrintEvent` we were hitting this `DCHECK` in `DispatchBeforePrintEvent`
https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/frame/web_local_frame_impl.cc;l=1867

if we were quickly calling `GetPageContent` and navigating.

Upstream, they mostly do this:
```cpp
DispatchBeforePrintEvent();

Print(...);

DispatchAfterPrintEvent();
```
but there are a bunch of different ways they invoke `Print` so wanted to get your eyes on this.

Reply:
This is alarming that we wasted cycle to call `GetPageContent` just to ignore the result (without waiting for it) and then trigger another `GetPageContent`

Reply:
as discussed in DM, the timing we call GetPageContent in AddContent is before pdf even loaded so it triggered unintended print preview for `AIChatUIBrowserTest.FetchPageContentPDF`. 
We can add `!is_pdf_` for `DispatchBeforePrintEvent` since `delegate_->GetPdfElement` get nullptr in this case and we don't want to fire `beforeprint` when failing to recognize pdf